packages/template-retail-react-app/app/ssr.js

```diff
@@ -63,7 +63,8 @@ const {handler} = runtime.createHandler(options, (app) => {
     })
     app.get('/robots.txt', serveStaticFile('static/robots.txt'))
 
-    runtime.addSSRRenderer(app)
+    app.get('/worker.js(.map)?', runtime.serveServiceWorker)
+    app.get('*', runtime.render)
```

Comment:
@kevinxh (cc @alexvuong ) Getting back to our conversation from yesterday: specifically, why is `serveServiceWorker` a method from the `runtime` object (while `serveStaticFile` on line 64 is not)? It's because:

- `serveServiceWorker` has different implementation depending on whether the runtime is [local dev](https://github.com/SalesforceCommerceCloud/pwa-kit/blob/eb86d1970912a230696aa7e4a4d9ae57408aa809/packages/pwa-kit-cli/src/ssr/server/build-dev-server.js#L151-L166) or [remote](https://github.com/SalesforceCommerceCloud/pwa-kit/blob/eb86d1970912a230696aa7e4a4d9ae57408aa809/packages/pwa-kit-runtime/src/ssr/server/build-remote-server.js#L643-L667) server
- `serveStaticFile` is a function that has the _same_ implementation regardless of local or remote server.

Reply:
From an end user's perspective, it would make sense to me that all the http serving functionalities are from the same place. I think we should design the API to be more user friendly as oppose to based on the actual implementation.

Reply:
> serveStaticFile is a function that has the same implementation regardless of local or remote serve

There are other things that are the same on dev/prod, too, and we still include these in the runtime. The decision to include it or not isn't based on this logic – it's simply that I didn't have time/feedback to move it.

I am with @kevinxh on this, and I had the same thought myself as I looked at it this week.

Reply:
On a similar note, IMO utilities like `isRemote` should be a part of the runtime API, what's your thought on moving `isRemote` to the runtime API? @olibrook

Reply:
The thing that we are missing is a way to make `RemoteServerFactory` and `DevServerFactory` to inherit a base class.

Reply:
Well, I would avoid opening a can of worms here. There are lots and lots of things that you could include. But it's a lil' late in the day. The `serveStaticFile` thing is dead obvious. I think `isRemote` is less obvious, because you want to use it in React code, too (and you typically don't access the runtime from React).

> The thing that we are missing is a way to make RemoteServerFactory and DevServerFactory to inherit a base class, or an abstract factory.

We're not missing this – what we have here is something I arrived at after having tried those already and then very carefully pulling apart the dependencies to make sure we don't accidentally ship devServer to production.

I would (1) move `serveStaticFile` because it's easy and has nice aesthetics (2) ship V2 and (3) think about some simple API changes that you could do for a V3 release – we should get less sensitive about major version bumps.

@kevinxh Your time would be better spent doing that hot-reloading work you were excited about, no?

Reply:
All right, `serveStaticFile` has been moved.